### DISCUSSION TITLE
Is it possible to disable snippets?

### DISCUSSION THREAD

**Discussion by BenceBakos** [phpactor/phpactor#2625] [General]
Cheers!

I would like to turn off the following feature:

```php
        Log::
```
(auto completing `debug`)
```php
        Log::debug($message, $context)
```

I only want:
```php
        Log::debug()
```
(no parameter names)

Tried to turn it off with:
```lua
lspconfig.phpactor.setup({
	init_options = {
		["completion_worse.completor.worse_parameter.enabled"] = false,

	},
})
```

but it doesn't have any effect.

Also tried:
 - restarting lsp(`:LspRestart`)
 - `phpactor cache:clear` 

Any ideas how to turn this feature off? Maybe my configuration is wrong in some way.

**Comment by dantleech** [comment]
This setting does not relate tot this feature. The feature is "snippets" and it cannot be turned off on the phpactor side currently, but it might be possible to disable it on the client side somehow.

**Comment by przepompownia** [comment]
Try to update `capabilities` rather than `init_options` there. If for example you get `capabilities` from `vim.lsp.protocol.make_client_capabilities()`, try 
```lua
capabilities.textDocument.completion.completionItem.snippetSupport = false
```
before you pass it to `lspconfig.phpactor.setup`.

Edit: `vim.lsp.protocol.make_client_capabilities()` has `snippetSupport = false` by default so it's possible that you have changed it to true somewhere between.

**Comment by BenceBakos** [comment]
@przepompownia 
This turns off everything:
```lua
lspconfig.phpactor.setup({
	init_options = {
		["language_server_phpstan.enabled"] = false,
		["language_server_psalm.enabled"] = false,
	},

	capabilities = {
		textDocument = {
			completion = {
				completionItem = {
					snippetSupport = false
				}
			}
		}
	}
})
```


results in 
```php
Log::debug
```

but I'm hoping for:

```php
Log::debug()
```

**Comment by przepompownia** [comment]
From the Neovim perspective, look here https://github.com/phpactor/phpactor/pull/2527#issuecomment-1925442859 or directly to https://github.com/przepompownia/nvim-arctgx/blob/a36f84d599cd2c0f78860a7454ff8c38f274da28/lua/arctgx/pluginConfigs/nvim-cmp.lua#L139-L166

**Comment by przepompownia** [comment]
The example above is not about method but you can modify it to your needs. Maybe on Phpactor side it would be possible to provide more than one types of snippets but it's relatively simple to append parentheses on the client side.

**Comment by BenceBakos** [comment]
Yes, did just that. Thanks!

```lua
lspconfig.phpactor.setup({
	capabilities = {
		textDocument = {
			completion = {
				completionItem = {
					snippetSupport = false
				}
			}
		}
	}
})

local function confirmDone(evt)
  local context = evt.entry.context
  if context.filetype ~= 'php' then
    return
  end


  if vim.startswith(context.cursor_after_line, '(') then
    return
  end

  local endRange = evt.entry.source_insert_range['end']
  vim.treesitter.get_parser(context.bufnr):parse({endRange.line, endRange.line})
  local node = vim.treesitter.get_node({pos = {endRange.line, endRange.character - 1}})

  -- magic here
  local methodNodeTypes = {'class_constant_access_expression','member_access_expression'}

  if vim.tbl_contains(methodNodeTypes,node:parent():type()) then
	  vim.api.nvim_feedkeys('(', 'i', false)
  end

end

cmp.event:on('confirm_done', confirmDone)
```


This works for `Log::debug`, `$this->belongsTo` (appends brackets as I wished, but global method calls like `request` or `dd` does not seem to work.

When I log out `node:type()`, it sais "name". Not very usefull, but I can not see how treesitter could guess that it's a global method's name.

**Comment by przepompownia** [comment]
I forgot to add hint about `:InspectTree` in the case you are not familiar with it (assuming that you use master/nightly version and have installed treesitter for PHP (can be `php_only` too if you not inject HTML code)).

**Comment by BenceBakos** [comment]
Worked it out without that :D  
But now, I can confirm, without brackets, it does not recognises "request" as a "function_call_expression".
It just sais 
```
    (ERROR) ; [78:9 - 15]
     (name) ; [78:9 - 15]
```
